lib/rouge/lexers/livescript.rb

```diff
@@ -0,0 +1,305 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Livescript < RegexLexer
+      tag 'livescript'
+      aliases 'ls'
+      filenames '*.ls'
+      mimetypes 'text/livescript'
+
+      title 'LiveScript'
+      desc 'LiveScript, a language which compiles to JavaScript'
+
+      def self.detect?(text)
+        return text.shebang? 'lsc'
+      end
+
+      def self.declarations
+        @declarations ||= Set.new %w(const let var function class extends implements)
+      end
+
+      def self.keywords
+        @keywords ||= Set.new %w(
+          loop until for in of while break return continue switch case
+          fallthrough default otherwise when then if unless else throw try
+          catch finally new delete typeof instanceof super by from to til
+          with require do debugger import export yield
+        )
+      end
+
+      def self.constants
+        @constants ||= Javascript.constants + %w(yes no on off void)
+      end
+
+      def self.builtins
+        @builtins ||= Javascript.builtins + %w(this it that arguments)
+      end
+
+      loop_control_keywords = Set.new %w(break continue)
+      id = /[$a-z_]((-(?=[a-z]))?[a-z0-9_])*/i
+      int_number = /\d+(_\d+)*/
+      int = /#{int_number}(e-?#{int_number})?([A-Za-z_][A-Za-z0-9_]*)?/
+
+      state :root do
+        rule(%r(^(?=\s|/))) { push :slash_starts_regex }
+        mixin :comments
+        mixin :whitespace
+
+        # list of words
+        rule %r/(<\[)(.*?)(\]>)/m do
+          groups Punctuation, Str, Punctuation
+        end
+
+        # function declarations
+        rule %r/!\s*function\b/, Keyword::Declaration
+        rule %r/!?[-~]>|<[-~]!?/, Keyword::Declaration
+
+        # switch arrow
+        rule %r/(=>)/, Keyword
+
+        # prototype attributes
+        rule %r/(::)(#{id})/ do
+          groups Punctuation, Name::Attribute
+          push :id
+        end
+        rule %r/(::)(#{int})/ do
+          groups Punctuation, Num::Integer
+          push :id
+        end
+
+        # instance attributes
+        rule %r/(@)(#{id})/ do
+          groups Name::Variable::Instance, Name::Attribute
+          push :id
+        end
+        rule %r/([.])(#{id})/ do
+          groups Punctuation, Name::Attribute
+          push :id
+        end
+        rule %r/([.])(\d+)/ do
+          groups Punctuation, Num::Integer
+          push :id
+        end
+        rule %r/#{id}(?=\s*:[^:=])/, Name::Attribute
+        rule %r/(&)(#{id})?/ do
+          groups Name::Builtin, Name::Attribute
+        end
+
+        # operators
+        rule %r(
+          [+][+]|--|~|&&|\b(and|x?or|is(nt)?|not)\b(?!-[a-zA-Z]|_)|[?]|=|
+          [|]{2}|\.([|&^]|<<|>>>?)\.|\\(?=\n)|[.:]=|(<<|>>>?|==?|!=?|
```

Comment:
Would this be better to be consistent with the syntax used in the rest of the pattern?

```suggestion
          [|][|]|[.]([|&^]|<<|>>>?)\.|\\(?=\n)|[.:]=|(<<|>>>?|==?|!=?|
```

Reply:
Sure. To be honset, this was mainly copied from coffeescript lexer. Now I see some more problems with this regex:
1. there is no **`=** operator in LS (nor in CS)
2. there is no `>>>=` operator in LS
3. it splits `<<<` and `<<<<` into two operators. I can imagine this may produce unnecessary elements in Rouge's output.